### PR TITLE
ZS2022-14 Tests for create device

### PR DESCRIPTION
**Pull request by DariusChis**
https://zenitech.atlassian.net/jira/software/projects/ZS2022/boards/177?selectedIssue=ZS2022-14

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -26,7 +26,16 @@
             <artifactId>slf4j-simple</artifactId>
             <version>1.6.2</version>
             <scope>test</scope>
+
         </dependency>
+        <!-- https://mvnrepository.com/artifact/org.testng/testng -->
```
Comment (CaprarClaudiu) [inline pom.xml]: do we need this comment here?

File: src/main/java/pageObjects/CreateDevicePage.java [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+import driver.DriverFactory;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+
+import java.util.concurrent.TimeUnit;
+
+import static utils.Constants.*;
```
Comment (CaprarClaudiu) [inline src/main/java/pageObjects/CreateDevicePage.java]: try not to use wild imports. also, do we need the import to be static?

File: src/main/java/utils/Constants.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+public class Constants {
+
+    private static Random random = new Random();
+    public static final String LOGIN_PAGE_URL = "https://zenistock.zenitech.local/login";
+    public static final String ADMIN_USERNAME = "admin";
+    public static final String ADMIN_PASSWORD = "qazqwer1@d.";
+    public static final String DEVICE_NAME = "Laptop DELL";
+    public static final String DEVICE_SERIAL_NO = random.nextInt(100000) + "";
+    public static final String DEVICE_INVENTORY_NO = random.nextInt(100000) + "";
+    public static final String DEVICE_INVOICE_NO = random.nextInt(1000000) + "";
```
Comment (CaprarClaudiu) [inline src/main/java/utils/Constants.java:10]: do we need to concatenate the random number with an empty string?

File: src/main/java/pageObjects/CreateDevicePage.java [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+    public void createDeviceToLongInvoiceNo(){
+        devicesButton.click();
+        addDeviceButton.click();
+        nameField.sendKeys(DEVICE_NAME);
+        serialNoField.sendKeys(DEVICE_SERIAL_NO);
+        inventoryNoField.sendKeys(DEVICE_INVENTORY_NO);
+        invoiceNoField.sendKeys(DEVICE_NUMBER_OVER_30_CHAR);
+        saveDeviceButton.click();
+    }
+}
```
Comment (CaprarClaudiu) [inline src/main/java/pageObjects/CreateDevicePage.java:251]: are we able to create a device with empty date fields? (invoice date, warranty start/end). can we create devices using a end date earlier than start date?

File: src/test/java/pageObjects/BaseTest.java [hunk 5]
```diff
@@ -1,21 +1,44 @@
...
 import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
 import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.support.FindBy;
 import org.testng.annotations.AfterMethod;
 import org.testng.annotations.BeforeMethod;
 
+import java.util.concurrent.TimeUnit;
+
+import static utils.Constants.*;
```
Comment (CaprarClaudiu) [inline src/test/java/pageObjects/BaseTest.java]: try not to use wild imports

File: src/test/java/pageObjects/CreateDevicePageTests.java [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+import java.util.concurrent.TimeUnit;
+
+public class CreateDevicePageTests extends BaseTest {
+
+
+    @Test
+    public void successfulCreateDevice() {
+        createDevicePage.successfullyCreateDevice();
+        WebElement successToaster = driver.findElement(By.id("success_toaster"));
+        Assert.assertEquals(successToaster.getText(), "Item created successfully!");
```
Comment (CaprarClaudiu) [inline src/test/java/pageObjects/CreateDevicePageTests.java:17]: if we successfully create a device, could we also check if this device was created and is shown in the table?

File: src/main/java/utils/Constants.java (hunk 3 above)
Comment (DariusChis) [reply src/main/java/utils/Constants.java:10]: the "sendKeys" method doesn't acceot arguments of type long, and that is why I had to convert them to strings. This seemed the simplest option to do so

**Review by ScorusMihai — APPROVED** [review]

**Review by CaprarClaudiu — APPROVED** [review]
We will have to add the date field tests once they will be available on the web app.

**Pull request merged**